File: SourceCode/MobilePhoneSalesManagement/ServerApp.DAL/Models/SpecificationType.cs
namespace ServerApp.DAL.Models
{
    public class SpecificationType
    {
        public int SpecificationTypeId { get; set; } // Primary Key
        public string Name { get; set; } // Tên thông số (vd: Màu sắc, Kích thước)
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;

        // Navigation properties
        public virtual ICollection<ProductSpecification> ProductSpecifications { get; set; }
    }

}

File: SourceCode/MobilePhoneSalesManagement_Frontend/src/app/features/admin/specification-type-management/models/specificationType.ts
export interface specificationType {
    specificationTypeId: string;
    name: string;
    createdAt?: Date;
    updatedAt?: Date;
}

export interface productSpecifications {
    productId: string;
    specificationTypeId: string;
    value: string;
    createdAt: Date;
    updatedAt: Date;
    specificationType: specificationType;
}
